scripts/build_wheels.py

```diff
@@ -0,0 +1,65 @@
+import argparse
+import logging
+import os
+import subprocess
+import sys
+
+from _utils import _ensure_cwd_is_fairlearn_root_dir, _LogWrapper
+from process_readme import process_readme
+
+
+_logger = logging.getLogger(__file__)
+logging.basicConfig(level=logging.INFO)
+
+
+_fairlearn_dev_version_env_var_name = 'FAIRLEARN_DEV_VERSION'
```

Comment:
what I usually see is that the version is set in the main `__init__.py`. Any reason you have it set in the build process here?

Reply:
Because PyPI doesn't make upload and publish separate steps (each upload 'burns' the version number). This allows us to have multiple 'dev' versions on Test-PyPI and then send the final changeset to PyPI itself.

Reply:
I'll consider this closable unless there's any further suggestion on how to reconcile these goals @adrinjalali ?

Reply:
I'm not sure if I understand why you need to upload several copies of the same release to pypi. I haven't seen this issue since we upload only once per release.

Reply:
We would prefer to only upload once per release too. However, it's not unknown for us to discover bugs after uploading to Test-PyPI. We try to remedy issues which we find so that they cannot reoccur, but they can still elude us.

Reply:
ah I see. That's why we do at least one RC before each release, and if things go wrong and the release needs a quick update, we do either a post release if the changes are in the docs only, and if there are code changes, we do a quick minor release afterwards. [PEP440](https://www.python.org/dev/peps/pep-0440/) specifies how the version numbers should work, in case you haven't seen it.

Reply:
We use RC/alpha releases as well. @riedgar-ms maybe we're overdoing it here?

Reply:
There was a reason why I used `dev` instead of `rc` but I can't remember it now. It may have because `dev` could be appended to an `rc` but obviously on cannot stack up two `rc`s

Reply:
there's a clear distinction between `dev` and `rc` AFAIK. `dev` is what you have on master, and you usually don't upload them. You could have nightly builds for instance, with `devN` suffix, but the `rc` is for people out there to test the release, and it usually identifies a feature freeze for the release.

Reply:
Why would you have multiple indicators, @riedgar-ms ?

Reply:
That's the way the spec is written. I think (but I don't recall exactly) that the thing which steered me in that direction was that `dev` sorts at the lowest priority.

Reply:
FYI I consider this orthogonal to this PR, but we should have this discussion, potentially on a Teams thread.

Reply:
Or on an issue :P

Reply:
Sure

Reply:
https://github.com/fairlearn/fairlearn/issues/258